src/mantine-core/src/Rating/Rating.tsx

```diff
@@ -0,0 +1,302 @@
+    onMouseEnter,
+    onMouseMove,
+    onMouseLeave,
+    ...others
+  } = useComponentDefaultProps('Rating', defaultProps, props);
+
+  const { classes, cx, theme } = useStyles(
+    { size, readonly },
+    { name: 'Rating', classNames, styles, unstyled }
+  );
+
+  const _name = useId();
+  const rootRef = useRef<HTMLDivElement>(null);
+  const [stableValue, setStableValue] = useUncontrolled({
+    value,
+    defaultValue,
+    finalValue: 0,
+  });
+  const decimalUnit = 1 / fractions;
+
+  const [interactiveValue, setInteractiveValue] = useState(-1);
+  const [isClientOutside, setClientOutside] = useState(true);
+
+  const stableValueRounded = roundValueTo(stableValue, decimalUnit);
+  let _value = stableValueRounded;
+  if (interactiveValue !== -1) _value = interactiveValue;
+
+  const handleMouseEnter = (event: React.MouseEvent<HTMLDivElement>) => {
+    if (onMouseEnter) onMouseEnter(event);
+    setClientOutside(false);
+  };
+
+  const handleMouseMove = (event: React.MouseEvent<HTMLDivElement>) => {
+    if (onMouseMove) onMouseMove(event);
+    const { left, right, width } = rootRef.current.getBoundingClientRect();
+
+    const symbolWidth = width / count;
+
+    let hoverPosition;
+    if (theme.dir === 'rtl') hoverPosition = right - event.clientX;
+    else hoverPosition = event.clientX - left;
+
+    const hoverValue = hoverPosition / symbolWidth;
+
+    let rounded = roundValueTo(hoverValue + decimalUnit / 2, decimalUnit);
+
+    rounded = clamp(rounded, decimalUnit, count);
+
+    setInteractiveValue(rounded);
+    if (onChangeHover && rounded !== interactiveValue) onChangeHover(rounded);
+  };
+
+  const handleMouseLeave = (event: React.MouseEvent<HTMLDivElement>) => {
+    if (onMouseLeave) onMouseLeave(event);
+    setInteractiveValue(-1);
+    setClientOutside(true);
+    if (onChangeHover && interactiveValue !== -1) onChangeHover(-1);
+  };
+
+  const handleBlur = () => {
+    if (isClientOutside) {
+      setInteractiveValue(-1);
+    }
+  };
+
+  // const handleFocus = (event: React.FocusEvent<HTMLInputElement>) => {
+  // const resultedValue = parseFloat(event.currentTarget.value);
+  // setInteractiveValue(resultedValue);
+  // };
+
+  const handleClick = (event: React.MouseEvent<HTMLInputElement>) => {
+    if (event.clientX === 0 && event.clientY === 0) {
+      // called by keyboard event
+      const resultedValue = parseFloat(event.currentTarget.value);
+      setInteractiveValue(resultedValue);
+
+      return;
+    }
+
+    setStableValue(0);
+    setInteractiveValue(-1);
+    if (onChange && parseFloat(event.currentTarget.value) === stableValueRounded) {
+      onChange(0);
+    }
+  };
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const resultedValue = parseFloat(event.target.value);
+
+    setStableValue(resultedValue);
+    if (onChange) {
+      onChange(resultedValue);
+    }
+  };
+
+  return (
+    <Box
+      aria-label={`${stableValueRounded} star rating`}
```

Comment:
Remove `aria-label` – in this case it will not support localization. It should be defined on user side.